### ISSUE TITLE
Filter, search and sort users on responsiveness, feedback, etc. (Map AND Search)

### ISSUE THREAD

**Issue by rfay** [warmshowers/Warmshowers.org#340]
Over in https://www.warmshowers.org/node/43124 (excellent thread) it was suggested to allow filtering/searching/sorting by feedback and responsiveness, great ideas I think. Except that it's my opinion that most people use the map for most things... However, the map view _could_ easily filter by these things.

Another metric that has been suggested is last login.

**Comment by kalpaitch** [comment]
Adding member feedback on 3/82015 relating to improvements needed in default text search -

"Hi. I just got a request for a warm showers stay, but we should be listed as unavailable at this time. I checked my page and it says that I will not currently show up on the map. Do you know how they contacted me?"

She is of course right, but she showed up easily enough on the text search.

**Comment by klzig** [comment]
@kalpaitch , since we already scrub the contact info for unavailable hosts, it seems like the only way of contacting such a host (without prior knowledge of their contact info) is through the Send Message button.  Maybe the solution is as simple as popping up a warning when this button is pressed for an unavailable host.  But In general it seems like the WS site already goes out of its way to make this problem unlikely.

I'd hate to see us stop returning unavailable hosts in response to a search.  Search is the best (and possibly only) means of locating a member to give feedback, etc. if their username is lost.

**Comment by kalpaitch** [comment]
@klzig Fair point. I think my issue is that we have two search mechanisms which work differently to each other, this is what I'd like to address.

I like you're suggestion on the 'Send Message' button, I was thinking something similar also. Will put a separate ticket in.

**Comment by GeertDD** [comment]
+1 for this feature! It would make searching for a good host so much easier, especially in cities. I hope to see some progress on this issue soon. :grin:

**Comment by kalpaitch** [comment]
Options to implement user profile completion percentage into the search result weighting as well - https://github.com/warmshowers/Warmshowers.org/issues/748

**Comment by CyrilWendl** [comment]
With respect to user friendliness, I guess the best think wouldn't opt for filtring options, but even cooler would be a color or number of points / stars showing on every map, serving as a proxy for the users activities - responsiveness, feedback, profile completeness (see kalpaitch's linked issue). That would allow you seeing on which points you prefer to click even before having to check any boxes or else at al.
